.github/workflows/master_dev_checks.yml

```diff
@@ -0,0 +1,409 @@
+                  console.log(existing_versions)
+                  const version = file_entry.filename.split("__")[0].split("/").pop();
+                  console.log(version)
+                  console.log(existing_versions)
+                  if (existing_versions.includes(version)) {
+                    console.log(file_entry.filename);
+                    core.setFailed(`Flyway file with version ${version} already exists`);
+                  }
+                }
+              });
+  
+        - name: check for terraform (code)
+          if : github.event_name == 'pull_request'
+          id: terraform_check
+          uses: actions/github-script@v6.3.3
+          with:
+            github-token: ${{ secrets.CIPIPELINE_NEXUS_PASSWORD  }}
+            script: |
+              head_commit = "${{ github.sha }}"
+              base_commit = "${{ github.sha }}"
+              if ( "${{ github.event.before }}" != '0000000000000000000000000000000000000000') { 
+                base_commit = "${{ github.event.before }}"
+              }
+              if ( "${{ github.event_name }}" == 'pull_request') {
+                console.log("Using the base branch's commit for comparing.")
+                base_commit = "${{ github.event.pull_request.base.sha  }}"
+              }
+              console.log("base_commit " + base_commit);
+              console.log("head_commit " + head_commit); 
+              response = await github.rest.repos.compareCommits({
+                  owner: "${{ github.repository_owner }}",
+                  repo: "${{  github.repository }}".split('/')[1],
+                  head: head_commit,
+                  base: base_commit
+              });
+              var terraform_file = [] 
+              response.data.files.forEach(function(file_entry) {
+                if (file_entry.filename.endsWith(".tf") && file_entry.status != 'removed') {
+                  console.log(file_entry.filename);
+                  console.log(file_entry.status);
+                  terraform_file.push(file_entry.filename)
+                  console.log(terraform_file) ;
+                }
+              });
+              core.setOutput('terraform_file' , terraform_file.join(','));
+
+        - name: Check Version sequence of DB Scripts
+          id: check_db_version_seq
+          if: steps.check_db_migration.outputs.db_migration_dir
+          run:
+            python3 .github/scripts/check_db_migration_version_sequence.py "${{ steps.check_db_migration.outputs.db_migration_dir }}"      
+        
+  
+        - name: check for terraform version
+          if: steps.terraform_check.outputs.terraform_file 
+          env:
+            terraform_file: ${{ steps.terraform_check.outputs.terraform_file }}
+          run: |
+            python3 - <<EOF
+            import os
+            terraform_file = os.environ.get('terraform_file')
+            version_missing = False
+            for terraform_file in terraform_files.split(','):
```

Comment:
This code has a bug. How was this tested?

Reply:
Noticed and fixed the bug, this code I just copied from existing harness code written before me